service/dap/server_test.go

```diff
@@ -5353,11 +5353,37 @@ func TestNoDebug_AcceptNoRequestsButDisconnect(t *testing.T) {
 
 		// Disconnect request is ok
 		client.DisconnectRequestWithKillOption(true)
-		client.ExpectOutputEventTerminating(t)
-		client.ExpectOutputEventRegex(t, fmt.Sprintf(daptest.ProcessExited, "(-1|1)"))
-		client.ExpectTerminatedEvent(t)
-		client.ExpectDisconnectResponse(t)
-		client.ExpectTerminatedEvent(t)
+		terminated, disconnectResp := false, false
+		for {
+			m, err := client.ReadMessage()
```

Comment:
You can use ExpectMessage helper to skip the error check.

Reply:
ExpectMessage will error when it can't read a message, which I don't want.

Reply:
Why not?

Reply:
Because I want to read all messages until the connection is closed. Also I've added a default clause below that errors on unexpected messages.